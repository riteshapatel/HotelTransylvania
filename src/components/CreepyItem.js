/**
 * @class CreepyItem component 
 * @author ritesh.patel
 */
import React, {Component} from 'react';
import {Text, View, Image} from 'react-native';
import styles from '../styles/Global.style.js';
import CreepyButton from './CreepyButton';

export default class CreepyItem extends Component {
    render () {        
        const data = this.props.data || {};    
        return (
            <View style={styles.creepy_item}>
                <Image source={require('../assets/mummy_small.png')} />
                <View>
                <Text style={styles.item_name}>{`${this.props.hotelName}`}</Text>
                <Text style={styles.item_small}>{`\nName: ${this.props.data.name}\n`}{`From: ${this.props.data.arrivalDate}\n`}{`To: ${this.props.data.departureDate}\n`}</Text>
                </View>
                <View>
                <CreepyButton title='Reserve Coffin' navigateTo='Add' navigate={this.props.navigate} data={data}/> 
                <CreepyButton title='View Coffin' navigateTo='View' navigate={this.props.navigate} data={data}/>                                                     
                </View>
            </View>                
        )
    }
}