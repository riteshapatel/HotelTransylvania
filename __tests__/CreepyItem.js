/**
 * Unit test for CreepyItem component 
 * @author ritesh.patel
 */
import React from 'react';
import CreepyItem from '../src/components/CreepyItem';
import {MockedProvider} from 'react-apollo/test-utils';

import renderer from 'react-test-renderer';

test ('snapshot CreepyItem component', () => {
    const data = {};
    const tree = renderer.create(<MockedProvider><CreepyItem data={data} /></MockedProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});