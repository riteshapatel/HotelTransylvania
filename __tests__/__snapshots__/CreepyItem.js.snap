// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot CreepyItem component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F3D9EA",
      "borderColor": "#CCC",
      "borderWidth": 1,
      "color": "white",
      "flexDirection": "row",
      "fontSize": 14,
      "fontWeight": "bold",
      "justifyContent": "space-between",
      "margin": 2,
      "padding": 10,
      "textAlign": "left",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/mummy_small.png",
      }
    }
  />
  <View>
    <Text
      style={
        Object {
          "color": "#650444",
          "fontWeight": "bold",
        }
      }
    >
      undefined
    </Text>
    <Text
      style={
        Object {
          "fontSize": 10,
          "fontStyle": "italic",
        }
      }
    >
      
Name: undefined

      From: undefined

      To: undefined

    </Text>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#640544",
          "borderColor": "#404040",
          "borderRadius": 4,
          "borderWidth": 1,
          "marginBottom": 5,
          "opacity": 1,
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Reserve Coffin
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#640544",
          "borderColor": "#404040",
          "borderRadius": 4,
          "borderWidth": 1,
          "marginBottom": 5,
          "opacity": 1,
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        View Coffin
      </Text>
    </View>
  </View>
</View>
`;
